### ISSUE TITLE
Plain SQL INSERT .. RETURNING doesn't work with unqualified field templates in SQL Server

### ISSUE THREAD

**Issue by ergonmichelle** [jOOQ/jOOQ#14215]
### Your question

We have problems with returning inserted data after an insert statement using SQL Server. 

Given the following test setup, we do not manage to add a returning clause to an insert statement:
```
Table myTable = table("my_table");
Field<Long> id = field("id", SQLDataType.BIGINT.identity(true));
Field<String> name = field("name", SQLDataType.NVARCHAR.nullable(false));
dslContext
	.createTable(myTable)
	.column(id)
	.column(name)
	.execute();
```

The following returns null: 
```
Record record = dslContext
	.insertInto(myTable, name)
	.values("a name")
	.returning()
	.fetchOne();
```

The following throws an exception (`org.jooq.exception.DataAccessException: SQL [declare @result table ([alias_3355] bigint); insert into my_table (name) output id into @result values (?); select [alias_3355] from @result [r];]; Invalid column name 'id'.`):
```
Record record = dslContext
	.insertInto(myTable, name)
	.values("a name")
	.returning(id)
	.fetchOne();
```

Also trying to return the `name` column instead of the identity column or trying to use `returningResult()` instead of `returning()` does not make a difference. 

We do not have any problems with the same code when using a Postgres database.

After reading this [blogpost](https://blog.jooq.org/the-many-ways-to-return-data-from-sql-dml/) and the javadoc of [InsertReturningStep](https://www.jooq.org/javadoc/3.17.1/org.jooq/org/jooq/InsertReturningStep.html) I believe that this should be possible, also with SQL Server, although it might require a second statement for non-identity columns. 

Is this a bug or are we using the API incorrectly?

### jOOQ Version

jOOQ Porfessional Edition 3.17.1

### Database product and version

Microsoft SQL Server 2019 (RTM-CU8-GDR) (KB4583459) - 15.0.4083.2 (X64) Nov 2 2020 18:35:09 Copyright (C) 2019 Microsoft Corporation Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2019 Standard 10.0 (Build 17763: ) (Hypervisor)

### Java Version

openjdk version "11.0.8" 2020-07-14 LTS

### JDBC driver name and version (include name if unofficial driver)

com.microsoft.sqlserver:mssql-jdbc:10.2.0.jre11

**Comment by lukaseder** [comment]
Thanks a lot for your message.

There are two usage issues here:

> The following returns null: [ ... ]

It might be possible to implement the syntax in SQL Server using the `OUTPUT inserted.*` clause. This would be similar to `select().from("some plain SQL table")` producing a `SELECT *` query.

I'll investigate this: https://github.com/jOOQ/jOOQ/issues/14216

> The following throws an exception

That seems more subtle. In SQL Server, column references in `OUTPUT` must be qualified with the `INSERTED` pseudo table (or `DELETED` in `UPDATE`, `DELETE`, `MERGE` statements). This isn't being done in this particular case, as you're using the plain SQL API (`DSL.table(String)`) instead of creating tables from identifiers:

- https://www.jooq.org/doc/latest/manual/sql-building/plain-sql/
- https://www.jooq.org/doc/latest/manual/sql-building/names/

E.g. this works for me:

```java
// Note the my_table qualification
Field<Long> id = field(name("my_table, "id"), SQLDataType.BIGINT.identity(true));
Field<String> name = field(name("my_table", "name"), SQLDataType.NVARCHAR.nullable(false));

// [...]

Record record = dslContext
	.insertInto(myTable, name)
	.values("a name")
	.returning(id)
	.fetchOne();
```

Clearly a usability problem which ought to be addressed. A lot of people use the plain SQL API when they mean to create identifiers using `name()`. Behind the scenes, these aren't the same thing, but for users, the API looks too similar.

I'll have to think about how to fix this properly. At least, you have a workaround for now.

> We do not have any problems with the same code when using a Postgres database.

That's because PostgreSQL allows for using unqualified identifiers in the `RETURNING id` clause, whereas `OUTPUT id` is invalid in SQL Server.

**Comment by lukaseder** [comment]
Also related, using an explicit asterisk: https://github.com/jOOQ/jOOQ/issues/14217

**Comment by lukaseder** [comment]
I'll make this bug here about the unqualified field templates that don't work in SQL Server (because of the lacking `INSERTED.*` qualification), but do work elsewhere.

Throughout the jOOQ API, for certain `field(String)` templates (those whose "templates" are just an identifier), the behaviour should be the same as that of `field(unquotedName(String))`

**Comment by lukaseder** [comment]
Related issues:

- https://github.com/jOOQ/jOOQ/issues/8917
- https://github.com/jOOQ/jOOQ/issues/13104

**Comment by lukaseder** [comment]
The usability problem exposed by this issue here will be further investigated: https://github.com/jOOQ/jOOQ/issues/14218

It's probably reasonable that for identifier-only usages of `field(String)`, users would like to receive the exact same behaviour as if they had written `field(unquotedName(String))`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0 and 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14219). Thanks again for your report!

The fix does not yet include:

- https://github.com/jOOQ/jOOQ/issues/14216
- https://github.com/jOOQ/jOOQ/issues/14217

**Comment by ergonmichelle** [comment]
Thanks very much for the quick reply, workaround and (partial) fix!
